db/new_seeds/scenarios/participants/ects/ect.rb

```diff
@@ -14,25 +14,25 @@ def initialize(school_cohort:, full_name: nil, email: nil)
 
           def build(**_profile_args)
             @user = FactoryBot.create(:seed_user, **new_user_attributes)
-
-            @participant_identity = FactoryBot.create(:seed_participant_identity, user:)
-
             @teacher_profile = FactoryBot.create(:seed_teacher_profile, user:, school: school_cohort.school)
-
-            @participant_profile = FactoryBot.create(:seed_ect_participant_profile, participant_identity:, teacher_profile:, school_cohort:)
+            @participant_profile = FactoryBot.create(:seed_ect_participant_profile,
+                                                     participant_identity: FactoryBot.create(:seed_participant_identity, user:),
+                                                     teacher_profile:,
+                                                     school_cohort:)
 
             self
           end
 
-          def chain_add_induction_record(**induction_args)
+          def with_induction_record(**induction_args)
             add_induction_record(**induction_args)
             self
           end
 
-          def add_induction_record(induction_programme:, start_date:, end_date:, induction_status: "active", training_status: "active")
+          def add_induction_record(induction_programme:, mentor_profile: nil, start_date: 6.months.ago, end_date: nil, induction_status: "active", training_status: "active")
```

Comment:
I think the convetion of `add_` for non-chainable and `with_` for chainable makes sense, let's continue that.